index.js

```diff
@@ -39,6 +35,14 @@ module.exports = {
     }
 
     this._terserOptions = Object.assign({}, defaultOptions, addonOptions);
+
+    if (
```

Comment:
Executing this check after the `Object.assign` logic makes sure the `sourcemaps` option is still respected, even when the user provides their own `terser` config.